packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -38,19 +39,34 @@ export default util.createRule({
       description:
         'Enforce using concise optional chain expressions instead of chained logical ands, negated logical ors, or empty objects',
       recommended: 'strict',
+      requiresTypeChecking: true,
```

Comment:
In retrospect, this feels like one of those times that an enum is better than a boolean.
It doesn't require type-checking all the time, right?

Reply:
Most rules that claim to require type checking only do so some of the time. For example, `no-for-in-array` only comes into play if you have `for...in` loops -- which most files don't have most of the time.